### PR TITLE
[NB] dont combine scalar product with other scalar op in multaries

### PR DESCRIPTION
**Pull request by kabdelhak**

### CODE REVIEW

File: OMCompiler/Compiler/NFFrontEnd/NFSimplifyExp.mo [hunk 1]
```diff
@@ -1594,7 +1594,7 @@ algorithm
     then result;
 
     // case 2.0 binary SOME(op) inverse operator (-, :)
-    case (SOME(op), Expression.BINARY()) guard(Operator.isCombineable(op, exp.operator)) algorithm
+    case (SOME(op), Expression.BINARY()) guard(Operator.isCombineable(op, exp.operator) and not Operator.isScalarProduct(exp.operator)) algorithm
```
Comment (phannebohm) [inline OMCompiler/Compiler/NFFrontEnd/NFSimplifyExp.mo:1597]: Why not change the definition of `isCombinable`?

Comment (kabdelhak) [reply OMCompiler/Compiler/NFFrontEnd/NFSimplifyExp.mo:1597]: yes i could do that, neither of the operators should be scalar product i guess

File: OMSimulator
Comment (phannebohm) [inline OMSimulator:1]: I think you accidentally added a submodule change.

**Review by phannebohm — CHANGES_REQUESTED** [review]
Please revert the OMSimulator change.

**Comment by kabdelhak** [comment]
> Please revert the OMSimulator change.

i did! thanks for pointing it out

**Pull request merged**